transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -546,6 +554,11 @@ protected void cleanup() {
                 logger.warn("Failed to close the epoll fd.", e);
             }
             try {
+                // Ensure any inflight wakeup writes have been performed prior to closing eventFd.
+                // Cap number of iterations to avoid indefinite spin in the case of a failed write.
```

Comment:
This doesn't end up waiting for the write at all. Is the looping just basically having a sleep() and hoping for the best?

Reply:
@ejona86 do you mean in the case the "safeguard" limit is reached? Given syscall latency is in the order of microsecs, we would have to have waited 10's of millisecs for concurrent writes to complete, which I thought would be long enough to assume they did really complete and something else unexpected must have happened instead (i.e. some prior write failed for whatever reason).

Reply:
Oh also maybe I should add some comments to make this part clearer too: in almost all cases only a single iteration (single read) will be performed, i.e. first value of eventfd read will already `== eventFdWriteCount`. The only times it won't will be in a race situation of the kind that we're trying to avoid.

Reply:
Isn't that assuming that the other thread is running on a different core? If the wakeup-triggering code is running on the same core as this code but was context switched out before doing the write, then this code could run with a full timeslice before the wakeup code resumes.

Reply:
Fair enough, maybe that's a good reason to go for the `epoll_wait` option